test/hotspot/jtreg/compiler/codecache/CheckLargePages.java

```diff
@@ -111,8 +111,20 @@ private static void testNonSegmented1GbCodeCacheWith1GbLargePages() throws Excep
         out.shouldNotContain("CodeHeap 'profiled nmethods'");
         out.shouldNotContain("CodeHeap 'non-profiled nmethods'");
         out.shouldContain("UseLargePages=1, UseTransparentHugePages=0");
-        out.shouldMatch("CodeCache:  min=1[gG] max=1[gG] base=[^ ]+ size=1[gG] page_size=1[gG]");
+        out.shouldMatch("Large page support enabled\\. Usable page sizes: .*1[gG].*\\. Default large page size: .*\\.");
+        out.shouldMatch(
+                // 1GB large pages configured and available
+                "CodeCache:\\s+min=1[gG] max=1[gG] base=[^ ]+ size=1[gG] page_size=1[gG]|" +
+                // 1GB large pages configured but none available
+                "Failed to reserve and commit memory with given page size\\. " +
+                "req_addr: [^ ]+ size: 1[gG], page size: 1[gG], \\(errno = 12\\)");
```

Comment:
Took me a while to figure that these are `OR` matches due to the `|` hiding at the end of the first line. Would it make sense to update the comment to something like this?
```
                // 1GB large pages configured and available
                "CodeCache:\\s+min=1[gG] max=1[gG] base=[^ ]+ size=1[gG] page_size=1[gG]|" +
                // or 1GB large pages configured but none available
```

Reply:
Also, isn't there a `CodeCache:\\` line in the output in the failing case as well that should be added here in the OR part?

Reply:
Thanks @TobiHartmann for looking at it.
I've actually changed the test to follow @eastig's suggestion below and reverted these lines to their original state.